src/components/VmDetails/cards/SnapshotsCard/sagas.js

```diff
@@ -23,7 +23,7 @@ function* deleteVmSnapshot (action) {
   }
   yield put(addSnapshotRemovalPendingTask(snapshotId))
   let snapshotRemoved = false
-  for (let delaySec of [ 4, 4, 4, 30, 60 ]) {
+  for (let delaySec of [ 4, 4, 4, 30, 30, 60 ]) {
```

Comment:
Using sync requests for this type action is not very good idea, because deleting can last to long, thus even can be 'TIMEOUT'. It means that async way is more safe way to then sync. Also we can do some manipulation in delete action, while snapshot still deleting.

Reply:
The problem is not on client. Because after 14 minutes Engine drop connection by itself and return empty result.

Reply:
> The problem is not on client. Because after 14 minutes Engine drop connection by itself and return empty result.

That makes more sense.  The request would timeout on the server side.

Another question then - does the snapshot get put into a "delete in progress" state or anything useful like that?  If it did, you could at least filter out or distinctly display that snapshot...